opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java

```diff
@@ -255,16 +261,28 @@ public DiffData getDiffData() {
             InputStream[] in = new InputStream[2];
             try {
                 // Get input stream for both older and newer file.
+                ExecutorService executor = Executors.newFixedThreadPool(2,
```

Comment:
Is a private pool necessary? I wonder why not use `IndexerParallelizer` `getHistoryExecutor()` and `CountDownLatch` with timeout (and cancellation) to track the work?

Reply:
While I agree permanent pool (for the whole web app) would be better, I have several objections to using `IndexerParallelizer` pool:
  - nit: the name suggests it should be used for indexing
  - the pools do not name the threads (which I find useful when debugging) - this can be fixed though
  - the `lzHistoryExecutor` has number of threads set to `env.getHistoryParallelism()` which I am not sure translates well into web app runtime